cpu/atmega_common/getopt.c

```diff
@@ -0,0 +1,182 @@
+/*
+* Copyright (C) 2018 Viraj Sahai <vsahai@usc.edu, virajsahai32@gmail.com>
+*
+* This file is subject to the terms and conditions of the GNU Lesser General
+* Public License v2.1. See the file LICENSE in the top level directory for more
+* details.
+*/
+
+/**
+* @ingroup     cpu_atmega_common
+* @{
+*
+* @file        getopt.c
+* @brief       Implementation of getopt lib.
+*
+* This implements the getopt and getopt_r functionality. getopt_r has a few caveats, make sure to use it correctly.
+*
+* @author      Viraj Sahai <vsahai@usc.edu, virajsahai32@gmail.com>
+*
+*/
+
+#include <stdio.h>
+#include <string.h>
+#include "getopt.h"
+
+char* optarg;
+int optind = 1, opterr = 1, optopt;
+
+int getopt(int argc, char* const argv[], const char* optstring)
+{
+    char* loc_optstr = NULL, loc_argvptr = NULL;
+    char opt;
+    optarg = NULL;
+
+    if(optind < 1 || optind > argc)
```

Comment:
Lot's of style issues, you can run uncrustify to format your files with a RIOT compliant style.

Reply:
Will do! Thanks